### PR TITLE
move certificates to resources package

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
- Move certificates to resources package
- Add all possible DNS names to apiserver Certificate
- Move templateData struct and functions into dedicated file

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/resources/apiserver/kubelet-client-certificate.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package apiserver
+
+import (
+	"github.com/kubermatic/kubermatic/api/pkg/resources"
+	"github.com/kubermatic/kubermatic/api/pkg/resources/certificates"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// KubeletClientCertificate returns a secret with the etcd client certificate
```
Comment (alvaroaleman) [inline api/pkg/resources/apiserver/kubelet-client-certificate.go]: I'd argue it returns a secret with the Kubelet client certificate ;)

File: api/pkg/resources/certificates/root-ca.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+// RootCA returns a function to create a secret with a ca
+func RootCA(data *resources.TemplateData, existing *corev1.Secret) (*corev1.Secret, error) {
+	create := getCACreator(resources.CASecretName, fmt.Sprintf("root-ca.%s", data.Cluster.Address.ExternalName))
+
+	return create(data, existing)
+}
+
+// FrontProxyCA returns a function to create a secret with a ca
```
Comment (alvaroaleman) [inline api/pkg/resources/certificates/root-ca.go]: I must say I find it pretty confusing that we use the same CA for everything except for extension API, in other places where we could use a dedicated CA (e.G. apiserver -> etcd communication or OpenVPN) we re-use the existing root ca.

IMHO we should be consistent and either create as many dedicated CAs as possible OR use the same CA for everything

File: api/pkg/resources/openvpn/internal-client-certificate.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package openvpn
+
+import (
+	"github.com/kubermatic/kubermatic/api/pkg/resources"
+	"github.com/kubermatic/kubermatic/api/pkg/resources/certificates"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// InternalClientCertificate returns a secret with the etcd client certificate
```
Comment (alvaroaleman) [inline api/pkg/resources/openvpn/internal-client-certificate.go]: Does it? If yes, thats probably a bug ;)

**Review by alvaroaleman — CHANGES_REQUESTED** [review]
Some nits, the only real issue I have is with the inconsistency of using a new CA for extension API when we re-use the existing one everywhere else.

File: api/pkg/resources/certificates/root-ca.go (hunk 2 above)
Comment (mrIncompetent) [reply api/pkg/resources/certificates/root-ca.go]: We switched actually to a dedicated CA a few weeks ago as using the root-ca caused a lot of error messages in the apiserver logs.
Therefore i would move the discussion out of this PR.

This PR makes handling multiple CA's fairly easy, therefore i don't mind creating a dedicated CA for OpenVPN and etcd as well.

File: api/pkg/resources/apiserver/kubelet-client-certificate.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/resources/apiserver/kubelet-client-certificate.go]: You might be right :grimacing:

File: api/pkg/resources/openvpn/internal-client-certificate.go (hunk 3 above)
Comment (mrIncompetent) [reply api/pkg/resources/openvpn/internal-client-certificate.go]: Maybe it does something different than the comment tries to describe :grimacing:

File: api/pkg/resources/certificates/root-ca.go (hunk 2 above)
Comment (alvaroaleman) [reply api/pkg/resources/certificates/root-ca.go]: Okay, fine for me, I just want to have a consistent approach there :+1:

**Comment by mrIncompetent** [comment]
@alvaroaleman issues addressed with https://github.com/kubermatic/kubermatic/pull/1742/commits/8e262af14d8c384e5102efe2e8859684160d0c4a

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**